subprojects/core/src/main/java/org/gradle/cache/internal/GradleUserHomeCleanupService.java

```diff
@@ -77,15 +79,16 @@ public void stop() {
     }
 
     private boolean execute(MonitoredCleanupAction action) {
-        ProgressLogger progressLogger = startNewOperation(action.getClass(), action.getDisplayName());
-        try {
-            return action.execute(new DefaultCleanupProgressMonitor(progressLogger));
-        } finally {
-            progressLogger.completed();
-        }
-    }
+        return buildOperationRunner.call(new CallableBuildOperation<Boolean>() {
+            @Override
+            public Boolean call(BuildOperationContext context) throws Exception {
+                return action.execute(new DefaultCleanupProgressMonitor(context));
+            }
 
-    private ProgressLogger startNewOperation(Class<?> loggerClass, String description) {
-        return progressLoggerFactory.newOperation(loggerClass).start(description, description);
+            @Override
+            public BuildOperationDescriptor.Builder description() {
+                return BuildOperationDescriptor.displayName(action.getDisplayName());
+            }
+        });
     }
```

Comment:
💭 This is the other part of the main functional change, where we switch from using `ProgressLoggerFactory` to `BuildOperationRunner`.